app/services/user_seeder.rb

```diff
@@ -0,0 +1,122 @@
+class UserSeeder
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  def initialize(**options)
+    @csv_file = options[:csv_file]
+    file = File.read(csv_file)
+    @csv = CSV.parse(file, headers: true)
+    validate_csv_headers
+    @email_domain = options[:email_domain]
+    validate_email_domain
+    @deploy_env = options[:deploy_env] || LoginGov::Hostdata.env
+  rescue Errno::ENOENT
+    raise ArgumentError.new("#{csv_file} does not exist")
+  end
+
+  # Seed the users whose PII is specified in the CSV file. Raises an
+  # ArgumentError if an email address is already taken and uses a transaction to
+  # avoid partial completion.
+  #
+  # @return [Integer] the number of users created
+  def run
+    validate_environment
+
+    ActiveRecord::Base.transaction do
+      @csv.each_with_index do |row, i|
+        email = "user#{i}@#{email_domain}"
+        row['email_address'] = email
+        row['password'] = PASSWORD
+        ee = EncryptedAttribute.new_from_decrypted(email)
+
+        User.create!(email_fingerprint: ee.fingerprint) do |user|
+          codes = setup_user(user: user, ee: ee)
+          row['codes'] = codes.join('|')
+
+          personal_key = create_profile(user: user, row: row)
+          row['personal_key'] = personal_key
+        end
+      end
+    end
+
+    save_updated_csv
+
+    csv.length # the number of saved users
+  rescue ActiveRecord::RecordNotUnique
+    msg = "email domain #{email_domain} invalid - would overwrite existing users"
+    raise ArgumentError.new(msg)
+  end
+
+  private
+
+  PASSWORD = 'S00per Seekr3t'
+
+  PII_ATTRS = %w[
+    first_name
+    middle_name
+    last_name
+    dob
+    ssn
+    phone
+    address1
+    address2
+    city
+    state
+    zipcode
+  ].freeze
+
+  attr_reader :csv_file, :csv, :email_domain, :deploy_env
+
+  def validate_csv_headers
+    return if (PII_ATTRS - csv.first.to_h.keys).empty?
+
+    msg = "#{csv_file} must be a CSV file with headers #{PII_ATTRS.join(',')}"
+    raise ArgumentError.new(msg)
+  end
+
+  def validate_email_domain
+    hostname_regex = /^(([^\W_]|[^\W_][a-zA-Z0-9\-]*[^\W_])\.)*([^\W_]|[^\W_][A-Za-z0-9\-]*[^\W_])$/
+    return if hostname_regex.match?(email_domain)
+
+    raise ArgumentError.new("#{email_domain} is not a valid hostname")
+  end
+
+  def validate_environment
+    return unless %w[prod staging].include? deploy_env
```

Comment:
I think INT/sandbox is the only valid use-case right now. At some point we may want to use this in personal environments and add a feature flag. 

For now I recommend we whitelist:
return if deploy_env == 'int'

Reply:
We need to be able to test this locally too, I disagree with `== int`

Reply:
I personally feel pretty comfortable allowing this everywhere in the lower environment (and in dev, obviously). I'll leave it as-is for now (since it will also require manual intervention on our end to run it somewhere it shouldn't run).